Source/DafnyCore/AST/TokenNewIndentCollector.cs

```diff
@@ -0,0 +1,1580 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Dafny;
```

Comment:
Not blocking: could we still reduce the size of this file to <1K LOC by moving code to related classes?

Reply:
I did a best effort and moved 500 LoC outside of this class. Only 1.1K LOC remain, I think I'll stop there, as all the remaining are shared.

Reply:
Thanks! x